src/main/kotlin/uk/gov/justice/digital/hmpps/hmppstier/service/AdditionalFactorsForWomen.kt

```diff
@@ -66,25 +65,19 @@ class AdditionalFactorsForWomen(
     }
 
   private fun getArsonOrViolencePoints(convictions: Collection<Conviction>): Int =
-    if (convictions.flatMap { it.offences }
-      .map { it.offenceDetail }.any {
-        OFFENCE_CODES.contains(it.mainCategoryCode)
-      }
-    ) 2 else 0
-
-  private fun getSentenceLengthPoints(convictions: Collection<Conviction>): Int {
+    if (convictions.flatMap { it.offenceMainCategoryCodes }.any { OFFENCE_CODES.contains(it) }) 2 else 0
 
-    val custodialSentences = convictions.map { it.sentence }.filter { isCustodial(it) }
-    val custodialConvictions = convictions.filter { it.sentence in custodialSentences }
+  private fun getSentenceLengthPoints(convictions: Collection<Conviction>): Int =
+    if (convictions
+      .filter { it.sentence.isCustodial() }
+      .any { it.sentence.isTenMonthsOrOver() || it.isIndeterminate() }
+    ) 2 else 0
 
-    val longerThanTenMonths = custodialSentences.any {
-      it.startDate != null && it.expectedSentenceEndDate != null && sentenceTenMonthsOrOver(it)
-    }
-    val indeterminate = custodialConvictions.any { INDETERMINATE_SENTENCE == it.latestCourtAppearanceOutcome }
-    return if (longerThanTenMonths || indeterminate) 2 else 0
-  }
+  private fun Sentence.isTenMonthsOrOver(): Boolean =
```

Comment:
If we take this approach I'm not sure where these should live

Reply:
this and the one below could actually live on the object if you don't subscribe to having a simple domain

Reply:
they are to all intents and purposes on it at least as far as this class is concerned

Reply:
I can't easily see from this PR where isCustodial has ended up, but as it has two uses it should be on the Domain I think